go/grpcweb/websocket_wrapper.go

```diff
@@ -31,6 +34,36 @@ func newWebSocketResponseWriter(wsConn *websocket.Conn) *webSocketResponseWriter
 	}
 }
 
+func (w *webSocketResponseWriter) EnablePing(timeOutInterval time.Duration) {
+	if timeOutInterval < time.Second {
+		return
+	}
+	w.timeOutInterval = timeOutInterval
+	w.timer = time.NewTimer(w.timeOutInterval)
+	dispose := make(chan bool)
+	w.wsConn.SetCloseHandler(func(code int, text string) error {
+		close(dispose)
+		return nil
+	})
+	go w.ping(dispose)
+}
+
+func (w *webSocketResponseWriter) ping(dispose chan bool) {
+	if dispose == nil {
+		return
+	}
+	defer w.timer.Stop()
+	for {
+		select {
+		case <-dispose:
+			return
+		case <-w.timer.C:
```

Comment:
Capture the time here:
```suggestion
		case tickTime := <-w.ticker.C:
```